Engine/DataFeeds/Enumerators/Factories/BaseDataCollectionSubscriptionEnumeratorFactory.cs

```diff
@@ -53,24 +42,33 @@ public IEnumerator<BaseData> CreateEnumerator(SubscriptionRequest request, IData
             using (var dataCacheProvider = new SingleEntryDataCacheProvider(dataProvider))
             {
                 var configuration = request.Configuration;
-                var tradableDays = _tradableDaysProvider(request);
                 var sourceFactory = (BaseData)Activator.CreateInstance(request.Configuration.Type);
 
-                // Note: this enumerator factory is currently only used in backtesting with coarse data
-                // and has been updated to behave in the same way as in live trading
+                // we want the first selection to happen on the start time
+                // so we need the previous tradable day time, since coarse
+                // files are for each tradable date but emitted with next day time
+                var previousTradableDay = Time.GetStartTimeForTradeBars(
+                    request.Security.Exchange.Hours,
+                    request.StartTimeLocal,
+                    Time.OneDay,
+                    1,
+                    false);
+                var tradableDays = new[] { previousTradableDay }.Concat(request.TradableDays);
```

Comment:
Wondering if this previous day logic should be handled via the caller instead of inside here. What's the desired behavior when adding a universe _after_ the algorithm has initialized? (not sure if that's a supported case).

Reply:
Thought about this one for a while too, I think since this is a special behavior just for Coarse, better to handle it internally, normalizing the caller code.
> What's the desired behavior when adding a universe after the algorithm has initialized? (not sure if that's a supported case).

I'd be inclined to say its not a behavior we support at the moment - the objective is for coarse selection to happen right away 👍